examples/express/index.ts

```diff
@@ -0,0 +1,170 @@
+  const sessionId = crypto.randomUUID()
+  // Use search params to store state so other key-value pairs
+  // can be added easily
+  const state = new URLSearchParams({
+    icecream: iceCreamSelection,
+  })
+
+  // Generate a PKCE pair
+  const { codeChallenge, codeVerifier } = generatePkcePair()
+
+  const { url, nonce } = sgid.authorizationUrl({
+    // We pass the user's ice cream preference as the state,
+    // so after they log in, we can display it together with the
+    // other user info.
+    state: state.toString(),
+    codeChallenge,
+    // Scopes that all sgID relying parties can access by default
+    scope: ['openid', 'myinfo.name'],
+  })
+  sessionData[sessionId] = {
+    state,
+    nonce,
+    codeVerifier,
+  }
+  return res
+    .cookie(SESSION_COOKIE_NAME, sessionId, SESSION_COOKIE_OPTIONS)
+    .json({ url })
+})
+
+apiRouter.get('/callback', async (req, res): Promise<void> => {
+  const authCode = String(req.query.code)
+  const state = String(req.query.state)
+  const sessionId = String(req.cookies[SESSION_COOKIE_NAME])
+
+  const session = { ...sessionData[sessionId] }
+  // Validate that the state matches what we passed to sgID for this session
+  if (session?.state?.toString() !== state) {
+    res.redirect(`${frontendHost}/error`)
+    return
+  }
+
+  // Validate that the code verifier exists for this session
+  if (!session?.codeVerifier) {
+    res.redirect(`${frontendHost}/error`)
+    return
+  }
+
+  // Exchange the authorization code and code verifier for the access token
+  const { accessToken, sub } = await sgid.callback({
+    code: authCode,
+    nonce: session.nonce,
+    codeVerifier: session.codeVerifier
+  })
+
+  session.accessToken = accessToken
+  session.sub = sub
+  sessionData[sessionId] = session
+
+  // Successful login, redirect to logged in state
+  res.redirect(`${frontendHost}/logged-in`)
+})
+
+apiRouter.get('/userinfo', async (req, res) => {
+  const sessionId = String(req.cookies[SESSION_COOKIE_NAME])
+
+  // Retrieve the access token and sub
+  const session = sessionData[sessionId]
+  const accessToken = session?.accessToken
+  const sub = session?.sub
+
+  // User is not authenticated
+  if (session === undefined || accessToken === undefined || sub == undefined) {
+    return res.sendStatus(401)
+  }
+  const userinfo = await sgid.userinfo({
+    accessToken,
+    sub,
+  })
+
+  // Add ice cream flavour to userinfo
+  userinfo.data.iceCream = session.state?.get('icecream') ?? 'None'
+  return res.json(userinfo)
+})
+
+apiRouter.get('/logout', async (_req, res) => {
+  return res
+    .clearCookie(SESSION_COOKIE_NAME, SESSION_COOKIE_OPTIONS)
+    .sendStatus(200)
+})
+
+const initServer = async (): Promise<void> => {
+  try {
+    app.use(cookieParser())
+    app.use('/api', apiRouter)
+
+    app.listen(PORT, () => {
+      console.log(`Server listening on port ${PORT}`)
+      void open(`http://localhost:${PORT}`)
+    })
+  } catch (error) {
+    console.error(
```

Comment:
we are no longer fetching the static files right? so I don't think we need this message anymore